pkg/ddl/notifier/events.go

```diff
@@ -380,3 +381,44 @@ func NewFlashbackClusterEvent() *SchemaChangeEvent {
 		tp: model.ActionFlashbackCluster,
 	}
 }
+
+type jsonSchemaChangeEvent struct {
+	TableInfo            *model.TableInfo
+	OldTableInfo         *model.TableInfo
+	AddedPartInfo        *model.PartitionInfo
+	DroppedPartInfo      *model.PartitionInfo
+	ColumnInfos          []*model.ColumnInfo
+	OldTableID4Partition int64
+
+	Tp model.ActionType
+}
+
+// MarshalJSON implements the json.Marshaler interface.
+func (s *SchemaChangeEvent) MarshalJSON() ([]byte, error) {
```

Comment:
use 

```golang
type SchemaChangeEvent struct {
  inner jsonSchemaChangeEvent
}
```

so we don't need to copy all fields, while the `SchemaChangeEvent` still has private members.

Reply:
Oh, I tried, it really works. But if so, when we want to access the filed in `SchemaChangeEvent`, we need to use `s.inner.TableInfo` or `s.inner.Tp` to do this, the code becomes loner. It looks like a trade off, if we don't want to copy all field, we need more code access the field.Do you have any
advice on this？

Reply:
It's expected to use `s.inner.XXX` to access them. The main reason is if in future we add more fields, we may forget to copy them between `SchemaChangeEvent` and `jsonSchemaChangeEvent`. This tradeoff is also agreed in the pre-RFC document.